lib/internal/Magento/Framework/Phrase/Renderer/Placeholder.php

```diff
@@ -23,13 +23,16 @@ public function render(array $source, array $arguments)
         $text = end($source);
 
         if ($arguments) {
-            $placeholders = [];
-            foreach (array_keys($arguments) as $key) {
-                $placeholders[] = '%' . (is_int($key) ? strval($key + 1) : $key);
-            }
-            $text = str_replace($placeholders, $arguments, $text);
+            $placeholders = array_map([$this, 'keyToPlaceholder'], array_keys($arguments));
+            $pairs = array_combine($placeholders, $arguments);
+            $text = strtr($text, $pairs);
         }
 
         return $text;
     }
+
+    private function keyToPlaceholder($key)
```

Comment:
I did that initially, but the anonymous function didn't inherit any variables from its parent scope (no *use* directive) so it seemed like a needless performance overhead to instantiate a `Closure` every time `render()` is called.